libcontainer/cgroups/cgroups.go

```diff
@@ -27,27 +27,18 @@ type Manager interface {
 	// Destroys the cgroup set
 	Destroy() error
 
-	// The option func SystemdCgroups() and Cgroupfs() require following attributes:
-	// 	Paths   map[string]string
-	// 	Cgroups *configs.Cgroup
-	// Paths maps cgroup subsystem to path at which it is mounted.
-	// Cgroups specifies specific cgroup settings for the various subsystems
-
-	// Returns cgroup paths to save in a state file and to be able to
-	// restore the object later.
-	GetPaths() map[string]string
-
-	// GetUnifiedPath returns the unified path when running in unified mode.
-	// The value corresponds to the all values of GetPaths() map.
-	//
-	// GetUnifiedPath returns error when running in hybrid mode as well as
-	// in legacy mode.
-	GetUnifiedPath() (string, error)
+	// Path returns a cgroup path to the specified controller/subsystem.
+	// For cgroupv2, the argument is unused and can be empty.
+	Path(string) string
 
 	// Sets the cgroup as configured.
 	Set(container *configs.Config) error
 
-	// Gets the cgroup as configured.
+	// GetPaths returns cgroup path(s) to save in a state file in order
+	// to restore it later.
+	GetPaths() map[string]string
+
+	// GetCgroups returns the cgroup data as configured.
 	GetCgroups() (*configs.Cgroup, error)
 }
```

Comment:
```go
type ManagerV1 interface{
  Manager
  Paths() map[string]string
}

type ManagerV2 interface{
  Manager
  Path() string
}
```

Reply:
That was one of my ideas as well, yes, but it seems the approach in this  PR is better.

Problem with the above is we need an if/switch everywhere. With this PR the difference in code using it is minimized.

Reply:
Second problem with the approach above is the save state structure is different for v1 and v2, and that is another complication which is avoided by this PR quite elegantly.

Reply:
If we use the same interface for v1 and v, can we rename `GetPaths() map[string]string` to something like `StateJSON() (marshalled string, err error)` ?

Reply:
It will require merging the two json objects, as well as some additional steps in loadState(), which is not going to look nice I'm afraid.

I'll look into it.

Reply:
Just `State() map[string]string` maybe.

Reply:
Unfortunately we can't do it since there is one place where this GetPaths() is literally used as a set of paths:

https://github.com/opencontainers/runc/blob/64416d34f30eaf69af6938621137b393ada63a16/libcontainer/process_linux.go#L102..L103

(`struct setnsProcess` is defined at 
https://github.com/opencontainers/runc/blob/64416d34f30eaf69af6938621137b393ada63a16/libcontainer/process_linux.go#L102..L103)